### DISCUSSION TITLE
Fuzzy search và elasticsearch-analysis-vietnamese

### DISCUSSION THREAD

**Discussion by vituthanh** [duydo/elasticsearch-analysis-vietnamese#140] [General]
Cho em hỏi có ai từng sử dụng fuzzy search trong plugin elasticsearch-analysis-vietnamese không ạ?

**Comment by duydo** [comment]
> Cho em hỏi có ai từng sử dụng fuzzy search trong plugin elasticsearch-analysis-vietnamese không ạ?

Sử dụng bình thường như các analyzers khác thôi em. Em có vấn đề gì hay sao?

**Comment by vituthanh** [comment]
curl -X GET "http://localhost:9200/app-santhau-nha-thau/_search?pretty" -H 'Content-Type: application/json' -d'
{
  "_source": "ThongTinDoanhNghiep",
  "query": {
    "bool": {
      "must": {
		"fuzzy": {
		  "ThongTinDoanhNghiep.Ten_DayDu": {
			"value": "xay lap nam hai",
			"fuzziness": "auto"
		  }
		}
      },
      "filter": {
        "term": {
          "ThongTinDoanhNghiep.MaQuocGia.keyword": "VN"
        }
      }
    }
  }
}'
Em sử dụng query như thế này nhưng kết quả lại ra không đúng cho lắm, nếu anh từng sử dụng có thể suggest em câu query ổn hơn được không ạ. Em muốn kết quả là xây lắp đại nam

![image](https://github.com/duydo/elasticsearch-analysis-vietnamese/assets/29084948/b42dc496-7c59-47b5-8ebe-996bae43481a)

**Comment by duydo** [comment]
Em xem điều chỉnh tham số "fuzziness" cho phù hợp với yêu cầu nhé https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#fuzziness

**Comment by vituthanh** [comment]
Em chỉnh từ 0 tới 50 vẫn k có kết quả
